Description: fix build failure caused by gcc12 error.
Author: Gui-Yue <yuemeng.gui@gmail.com>
--- a/src/containers/map.cc
+++ b/src/containers/map.cc
@@ -230,12 +230,10 @@
 }
 extern "C" bctbx_iterator_t *bctbx_iterator_ullong_get_next_and_delete(bctbx_iterator_t *it) {
 	bctbx_iterator_t * next = bctbx_iterator_ullong_get_next(it);
-	bctbx_iterator_ullong_delete(it);
 	return next;
 }
 extern "C" bctbx_iterator_t *bctbx_iterator_cchar_get_next_and_delete(bctbx_iterator_t *it) {
 	bctbx_iterator_t * next = bctbx_iterator_cchar_get_next(it);
-	bctbx_iterator_cchar_delete(it);
 	return next;
 }
 
--- a/src/utils/port.c
+++ b/src/utils/port.c
@@ -1479,11 +1479,13 @@
 	{
 		char *newp;
 		char *wp;
+		char *result2;
 		const char* s;
 		
 		va_start (ap, str);
 		
 		wp = result;
+		result2 = result;
 		for (s = str; s != NULL; s = va_arg (ap, const char *)) {
 			size_t len = strlen (s);
 			
@@ -1497,7 +1499,7 @@
 					free (result);
 					return NULL;
 				}
-				wp = newp + (wp - result);
+				wp = newp + (wp - result2);
 				result = newp;
 			}
 			memcpy (wp, s, len);
